File: sol/solution/0001-0100/0020/Solution.scala
object Solution {
    import scala.collection.mutable
    def isValid(s: String): Boolean = {
        val stack = new Array[Char](s.length)
        val openBrackets = "({["
        var result = true
        var index = 0
        var stackIndex = 0
        while (index < s.length && result) {
            val currentBracket = s.charAt(index)
            if(currentBracket == '(') {
                stack(stackIndex) = ')'
                stackIndex += 1
            } else if (currentBracket == '{') {
                stack(stackIndex) = '}'
                stackIndex += 1
            } else if (currentBracket == '[') {
                stack(stackIndex) = ']'
                stackIndex += 1
            } else if(stackIndex > 0 && stack(stackIndex-1) == currentBracket) {
                stackIndex -= 1
            } else {
                result = false
            }
            index += 1
        }
        stackIndex == 0 && result
    }
}

File: sol/solution/0001-0100/0020/Solution.dart
class Solution {
  bool isValid(String s) {
    List<String> stack = [];
    for (var c in s.split('')) {
      if (c == '(' || c == '{' || c == '[') {
        stack.add(c);
      } else {
        if (stack.isEmpty) return false;
        var top = stack.removeLast();
        if ((c == ')' && top != '(') || 
            (c == '}' && top != '{') || 
            (c == ']' && top != '[')) {
          return false;
        }
      }
    }
    return stack.isEmpty;
  }
}